python/lsst/daf/butler/_butler.py

```diff
@@ -286,31 +288,37 @@ def _retrieve_dataset_type(self, name: str) -> DatasetType | None:
             return None
 
     @classmethod
-    def get_repo_uri(cls, label: str) -> ResourcePath:
+    def get_repo_uri(cls, label: str, default: ResourcePath | str | None = None) -> ResourcePath:
         """Look up the label in a butler repository index.
 
         Parameters
         ----------
         label : `str`
             Label of the Butler repository to look up.
+        default : `lsst.resources.ResourcePath` or `str`, optional
```

Comment:
I do wonder whether there is any use case for the second parameter not being identical to the first. Why would you ever have a default that differed? In which case I'm not sure we need this parameter and should either have a boolean that indicates whether to default to the original or not, or always do the defaulting.

Reply:
I do not think there is any other use case now, but I wanted to keep it general enough for any future use cases and also similar to other Python patterns where getters accept a default value.

Reply:
But I am also OK with the boolean flag instead of default. I already tried to always-default, but that breaks some unit tests in a bit of non-trivial way.

Reply:
Okay. I think I prefer boolean.